### DISCUSSION TITLE
better cookie support

### DISCUSSION THREAD

**Discussion by siddhantk232** [fastn-stack/ft-sdk#20] [Ideas]
- https://github.com/FifthTry/ft-sdk/pull/19#discussion_r1588949454
- https://github.com/FifthTry/ft-sdk/pull/19#discussion_r1588898350

Also, not happy with the `RC<RefCell<Vec<Cookie>>>` type for `In::set_cookies`. Should replace this with a blanket `CookieStore` and move the `In::add_cookie()` method there.
Functions that want to manipulate cookies should then take `CookieStore` and not the whole `In` this way we avoid unnecessary copy.

**Comment by amitu** [comment]
# IntoCookie 

```rs
impl In {
    // note that this is private
    fn add_cookie_raw(&self, s: String) {
         self.cookies.push(s)
    }

    pub fn add_cookie<K: AsRef<str>, V: AsRef<str>(&self, key: K, value: V) {}
}

pub trait IntoCookie {
    fn to_cookie(&self) -> String;
    fn add_cookie_into(in_: &In) {
        in_.add_cookie_raw(self.to_cookie())
    }
}

// someday if we decide to expose our own cookie struct
impl IntoCookie for OurCookie {}

// we will support cookie crate behind a feature flag
#[cfg(feature = "cookie")]
impl IntoCookie for cookie::Cookie {}
```
